src/Symfony/Component/Validator/Constraints/DateTimeValidator.php

```diff
@@ -29,7 +29,13 @@ public function validate($value, Constraint $constraint)
             throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\DateTime');
         }
 
-        if (null === $value || '' === $value || $value instanceof \DateTimeInterface) {
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if ($value instanceof \DateTimeInterface) {
+            @trigger_error(sprintf('Validating a \\DateTimeInterface with "%s" is deprecated since version 4.1 and will throw an exception in 5.0. Use "%s" instead.', DateTime::class, Type::class), E_USER_DEPRECATED);
```

Comment:
The suggested alternative could also be to remove the constraint if you always have a `\DateTimeInterface|null` in your property, not especially to add a `Type` constraint, as this constraint is a no-op for DateTimeInterface